src/render/components/workflows.js

```diff
@@ -23,44 +23,55 @@ const isReverse = (flowA, flowB) => {
-      continue;
-    }
-    //ignore duplicates
-    if (filteredFlows.find(other => flowsEqual(other, flow))) {
-      continue;
-    }
-    //we only want one list item for bi-directional flow pairs
-    const reverse = filteredFlows.find(other => isReverse(other, flow));
-    if (reverse) {
-      reverse.bidi = true;
+    if (isReverse(flow, flow)) {
       continue;
     }
 
-    filteredFlows.push(flow);
+    if (flow.edit && flow.using) {
+      if (flow.using == thisItem) {
+        pushLabeledFlow("edit", "Edit", flow.edit, itemAnchor(flow.edit));
+      } else {
+        pushLabeledFlow("edit-with", "Edit with", flow.using, itemAnchor(flow.using));
+      }
+    } else if (flow.from && flow.to && flow.using) {
+      //we only want one list item for bi-directional flow pairs
+      const hasReverse = workflows.find(other => !flowsEqual(flow, other) && isReverse(other, flow));
+      if (hasReverse) {
+        const otherItem = flow.to == thisItem ? flow.from : flow.to;
+        pushLabeledFlow(`bidi-${otherItem}`, `To/from&nbsp;${itemAnchor(otherItem)} with`, flow.using, itemAnchor(flow.using));
+      } else if (thisItem == flow.from) {
+        //use &nbsp; to join words for better appearance on narrow windows
+        pushLabeledFlow(`to-${flow.to}`, `To&nbsp;${itemAnchor(flow.to)} with`, flow.using, itemAnchor(flow.using));
+      } else if (thisItem == flow.to) {
+        pushLabeledFlow(`from-${flow.from}`, `From&nbsp;${itemAnchor(flow.from)} with`, flow.using, itemAnchor(flow.using));
+      } else if (thisItem == flow.using) {
+        pushLabeledFlow(`${flow.from}-to`, `${itemAnchor(flow.from)}&nbsp;to`, flow.to, itemAnchor(flow.to));
+      }
+    } else {
+      throw new Error(`Cannot render unhandled workflow for item '${thisItem}': ${JSON.stringify(flow)}`);
+    }
   }
 
   return detailsList(
     "Workflows",
-    filteredFlows.map(flow => {
-      if (flow.edit && flow.using) {
-        return flow.using == thisItem ?
-          `Edit ${itemAnchor(flow.edit)}` :
-          `Edit with ${itemAnchor(flow.using)}`;
-      } else if (flow.from && flow.to && flow.using) {
-        //use &nbsp; to join words for better appearance on narrow windows
-        if (thisItem == flow.from) {
-          return `${flow.bidi ? "To/from" : "To"}&nbsp;${itemAnchor(flow.to)} with&nbsp;${itemAnchor(flow.using)}`;
-        } else if (thisItem == flow.to) {
-          return `${flow.bidi ? "To/from" : "From"}&nbsp;${itemAnchor(flow.from)} with&nbsp;${itemAnchor(flow.using)}`;
-        } else if (thisItem == flow.using) {
-          return `${itemAnchor(flow.from)}&nbsp;${flow.bidi ? "to/from" : "to"} ${itemAnchor(flow.to)}`;
-        }
+    Object.values(labeledFlows).map(({base, items}) => {
+      items = Object.values(items);
+      if (items.length == 1) {
+        return `${base}&nbsp;${items[0]}`;
+      } else if (items.length <= 3) {
+        return `${base} ${items.join(", ")}`;
```

Comment:
We should just get rid of this condition for clarity

Reply:
New version without this:

![jms-after-after](https://user-images.githubusercontent.com/1519264/93297950-27f22080-f7a7-11ea-96d7-0aaa44b76e6d.png)

Reply:
Nice. This should be more mobile compatible too